build_tools/circle/build_doc.sh

```diff
@@ -2,18 +2,74 @@
 set -x
 set -e
 
-# Introspect the commit to know whether or not we should skip building the
+# Inspect the commit to know whether or not we should skip building the
 # documentation: a pull request that does not change any file in doc/ or
 # examples/ folder should be skipped unless the "[doc: build]" is found the
 # commit message.
-BUILD_DOC=`python build_tools/circle/check_build_doc.py`
-echo -e $BUILD_DOC
-if [[ $BUILD_DOC == "SKIP:"* ]]; then
-    touch ~/log.txt  # the "test" segment needs that file
+
+get_build_type() {
+	if [ -z "$CIRCLE_SHA1" ]
+	then
+		echo SKIP: undefined CIRCLE_SHA1
+		return
+	fi
+	commit_msg=$(git log --format=%B -n 1 $CIRCLE_SHA1)
+	if [ -z "$commit_msg" ]
+	then
+		echo QUICK BUILD: failed to inspect commit $CIRCLE_SHA1
+		return
+	fi
+	if [[ "$commit_msg" =~ '.*[doc skip].*' ]]
+	then
+		echo SKIP: [doc skip] marker found
+		return
+	fi
+	if [[ "$commit_msg" =~ '.*[doc build].*' ]]
+	then
+		echo BUILD: [doc build] marker found
+		return
+	fi
+	if [ -z "$CI_PULL_REQUEST" ]
+	then
+		echo BUILD: not a pull request
+		return
+	fi
+	git_range="origin/master...$CIRCLE_SHA1"
+	git fetch origin master >&2
+	filenames=$(git diff --name-only $git_range)
+	if [ -z "$filenames" ]
+	then
+		echo QUICK BUILD: failed to get changed filenames for $git_range
+		return
+	fi
+	if echo "$filenames" | grep -q -e ^examples/ -e ^doc/
+	then
+		echo BUILD: detected doc/ or examples/ filename modified in $git_range: $(echo "$filenames" | grep -e ^examples/ -e ^doc/ | head -n1)
+		return
+	fi
+	echo QUICK BUILD: no doc/ or examples/ filename modified in $git_range:
+	echo "$filenames"
+}
+
+touch ~/log.txt  # the "test" segment needs this file
```

Comment:
You can remove this since the test is now a dummy echo command and not `cat log.txt | grep Traceback`